### DISCUSSION TITLE
coverage file for sonarqube using .net 4.7.1

### DISCUSSION THREAD

**Discussion by vrunda** [coverlet-coverage/coverlet#1329] [Q&A]
I am able to generate .coverge file on tfs build server, but sonarqube does not display anything on code coverage yet.

Is there any way to generate covergae in Cobertura file format for .net 4.7 ?

**Comment by daveMueller** [comment]
Yes coverlet can create the coverage report in several output formats including cobertura. It depends a bit on which coverlet driver you are using. If you are using `coverlet.collector` you can specify the output format in a [runsettings](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md#advanced-options-supported-via-runsettings) file or pass it through the [command line](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md#passing-runsettings-arguments-through-commandline). If you are using `coverlet.msbuild` you can also just pass it through the [command line](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/MSBuildIntegration.md#coverage-output) as well as for  `coverlet.console` ([here](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/GlobalTool.md#coverage-output)).

**Comment by MarcoRossignoli** [comment]
There's also a merge/converted tool my MS to help this task https://learn.microsoft.com/en-us/dotnet/core/additional-tools/dotnet-coverage#dotnet-coverage-merge

**Comment by Bertk** [comment]
Closing this discussion since it’s no longer relevant to the current state of the project.
